propsigner/client.go

```diff
@@ -0,0 +1,66 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+)
+
+const (
+	maxResponseMessageSize = 100 << 10
+)
+
+// RequestSignatureV1 request a signature for a deal proposal to a remote wallet.
+func RequestSignatureV1(
+	ctx context.Context,
+	h host.Host,
+	authToken string,
+	proposal market.DealProposal,
+	remoteWallet peer.ID) (*crypto.Signature, error) {
+	proposalCborBytes := &bytes.Buffer{}
+	if err := proposal.MarshalCBOR(proposalCborBytes); err != nil {
+		return nil, fmt.Errorf("marshaling deal proposal to cbor: %s", err)
+	}
+
+	req := pb.ProposalSigningRequest{
+		AuthToken:            authToken,
+		FilecoinDealProtocol: filecoinDealProtocolV1,
+		Payload:              proposalCborBytes.Bytes(),
```

Comment:
In the `Payload` we send the CBOR encoded deal proposal that should be signed. 
CBOR encoding is the decided codec for deal proposal marshaling in Filecoin, so stick to that.